src/main/java/io/fabric8/elasticsearch/plugin/filter/FieldStatsResponseFilter.java

```diff
@@ -72,6 +73,17 @@ public void apply(final Task task, final String action, final ActionRequest requ
             @SuppressWarnings("unchecked")
             @Override
             public void onResponse(ActionResponse response) {
+                if(INDICES_FIELD_STATS_READ_ACTION.equals(action) && response instanceof FieldStatsResponse) {
+                    if(((FieldStatsResponse)response).getIndicesMergedFieldStats().size() == 0) {
+                        LOGGER.trace("Modifying the response to be {}", RestStatus.NOT_FOUND);
+                        Throwable err = new ElasticsearchException("The index is empty. This could mean data has not yet been collected.\n"
+                                + "You can use the Time Picker to change the time filter or select a higher time interval",
+                                RestStatus.NO_CONTENT);
+                        
+                        listener.onFailure(err);
```

Comment:
Should we be calling both 'onfailure' and 'onresponse' ?

Reply:
well I was not sure if the onresponse is triggered when the onfailure was pervious called, but I can change it, I wanted to avoid if-else chaos ;)

Reply:
Without actually testing I am unable to confirm or deny the code you have here.  I am just not familiar with it.